pkg/southbound/clientManager.go

```diff
@@ -153,26 +125,26 @@ func getCertPoolDefault() *x509.CertPool {
 }
 
 // CapabilitiesWithString allows a request for the capabilities by a string - can be empty
-func CapabilitiesWithString(target Target, request string) (*gpb.CapabilityResponse, error) {
+func (target Target) CapabilitiesWithString(ctx context.Context, request string) (*gpb.CapabilityResponse, error) {
```

Comment:
we should pass here a pointer (target *Target) not to pass the whole object

Reply:
We don't change the target fields in this function. Why do we need to define it as a pointer? I defined as a pointer in ConnectTarget since we change the fields. Am I missing something?

Reply:
I thought that always a pointer would be better than a full object even if we do not use it since it's more lightweight.

Reply:
Thanks for your response. That makes sense. Another reason that think of is the consistency. Typically, If we have one function which uses receiver pointer, it is recommended to define the other ones using pointers as well to see a predictable behavior. I will make the changes accordingly.